src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/CombinationController.php

```diff
@@ -124,48 +125,65 @@ public function bulkEditFormAction(int $productId): Response
      *
      * @param Request $request
      * @param int $productId
-     * @param int $combinationId
      *
      * @return JsonResponse
      */
-    public function bulkEditAction(Request $request, int $productId, int $combinationId): JsonResponse
+    public function bulkEditAction(Request $request, int $productId): JsonResponse
     {
-        try {
-            // PATCH request is required to avoid disabled fields to be forced with null values
-            $bulkCombinationForm = $this->getBulkCombinationFormBuilder()->getFormFor($combinationId, [], [
-                'method' => Request::METHOD_PATCH,
-                'product_id' => $productId,
-            ]);
-        } catch (CombinationNotFoundException $e) {
-            return $this->returnErrorJsonResponse(
-                ['error' => $this->getErrorMessageForException($e, $this->getErrorMessages($e))],
-                Response::HTTP_NOT_FOUND
-            );
+        $combinationIds = $request->request->get('combinationIds');
+        if (!$combinationIds) {
+            return $this->json([
+                'fatal' => true,
+                'errors' => $this->getFallbackErrorMessage('', 0, 'Missing combinationIds in request body'),
+            ], Response::HTTP_BAD_REQUEST);
         }
 
-        try {
-            $bulkCombinationForm->handleRequest($request);
-            $result = $this->getBulkCombinationFormHandler()->handleFor($combinationId, $bulkCombinationForm);
-
-            if (!$result->isSubmitted()) {
-                return $this->json(['errors' => [
-                    'form' => [
-                        $this->trans('No submitted data.', 'Admin.Notifications.Error'),
-                    ],
-                ]], Response::HTTP_BAD_REQUEST);
+        $combinationIds = json_decode($combinationIds);
+        $errors = [];
+        foreach ($combinationIds as $combinationId) {
+            try {
```

Comment:
I think this part could be optimized, for now the form is rebuilt for each provided combination but it's always the same data so no need to build the form in the loop it could be built only once outside of the loop